src/BenchmarkDotNet/Jobs/JobExtensions.cs

```diff
@@ -330,24 +330,35 @@ public static Job WithEnvironmentVariable(this Job job, [NotNull] string key, [N
         /// <param name="job"></param>
         /// <param name="packageName">The NuGet package name</param>
         /// <param name="packageVersion">The NuGet package version</param>
+        /// <param name="source">(optional)Indicate the URI of the NuGet package source to use during the restore operation.</param>
+        /// <param name="prerelease">(optional)Allows prerelease packages to be installed.</param>
         /// <returns></returns>
-        public static Job WithNuGet(this Job job, string packageName, string packageVersion) => job.WithCore(j => j.Infrastructure.NuGetReferences = new NuGetReferenceList(j.Infrastructure.NuGetReferences ?? Array.Empty<NuGetReference>()) { new NuGetReference(packageName, packageVersion) });
+        public static Job WithNuGet(this Job job, string packageName, string packageVersion, Uri source = null, bool prerelease = false) =>
+            job.WithCore(j => j.Infrastructure.NuGetReferences =
+                new NuGetReferenceList(j.Infrastructure.NuGetReferences ?? Array.Empty<NuGetReference>())
+                    {
+                        new NuGetReference(packageName, packageVersion, source, prerelease)
+                    });
 
         /// <summary>
         /// Runs the job with a specific NuGet dependency which will be resolved during the Job build process
         /// </summary>
         /// <param name="job"></param>
         /// <param name="packageName">The NuGet package name, the latest version will be resolved</param>
+        /// <param name="source">(optional)Indicate the URI of the NuGet package source to use during the restore operation.</param>
+        /// <param name="prerelease">(optional)Allows prerelease packages to be installed.</param>
         /// <returns></returns>
-        public static Job WithNuGet(this Job job, string packageName) => job.WithNuGet(packageName, string.Empty);
+        public static Job WithNuGet(this Job job, string packageName, Uri source = null, bool prerelease = false) =>
```

Comment:
I think that instead of having two overloads with optional parameters we should have a single method with all optional parameters at once place:

```diff
- public static Job WithNuGet(this Job job, string packageName, string packageVersion, Uri source = null, bool prerelease = false)
- public static Job WithNuGet(this Job job, string packageName, Uri source = null, bool prerelease = false)
+ public static Job WithNuGet(this Job job, string packageName, string packageVersion = null, Uri source = null, bool prerelease = false)
```

and `null` for `packageVersion` should mean no package version specified (currently we have `string.Empty` for `packageVersion` and `null` for `source` (consistency)